git/operations.go

```diff
@@ -284,24 +306,35 @@ func changed(ctx context.Context, workingDir, ref string, subPaths []string) ([]
 	return splitList(out.String()), nil
 }
 
-func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
-	if trace {
-		print("TRACE: git")
-		for _, arg := range args {
-			print(` "`, arg, `"`)
+// Whilst debugging or developing, you may wish to filter certain git commands out of the logs.
+// To filter out a certain git subcommand add it to `blacklistedGitCommands` inside this function.
+func isExemptedTraceGitCommand(command string) bool {
+	blacklistedGitCommands := []string{
+		"config",
+	}
+	for _, blacklistedGitCommand := range blacklistedGitCommands {
+		if blacklistedGitCommand == command {
+			return true
 		}
-		println()
 	}
+	return false
+}
+
+// execGitCmd runs a `git` command with the supplied arguments.
+func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
 	c := exec.CommandContext(ctx, "git", args...)
 
 	if config.dir != "" {
 		c.Dir = config.dir
 	}
 	c.Env = append(env(), config.env...)
-	c.Stdout = ioutil.Discard
+
+	commandOutput := &bytes.Buffer{}
+	c.Stdout = commandOutput
 	if config.out != nil {
-		c.Stdout = config.out
+		c.Stdout = io.MultiWriter(commandOutput, config.out)
```

Comment:
the prior logic was to use `ioutil.Discard`, which, to me, is functionally equivalent to writing to a `commandOutput` that's never used if `trace == false`.  I can write it that way, but this was the simpler route, from what I could tell.  I understand why it could look strange to default to writing to a trace output that is potentially never used and using `ioutil.Discard` is at least explicit in its intent.  Let me know what you would like me to do here and I'll happily do it.

Reply:
I removed this from this PR and submitted separately:
https://github.com/weaveworks/flux/pull/1844